### PR TITLE
MULE-14872: Document mule DataWeave functions and review variables doc

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

**Comment by afelisatti** [comment]
@dukesphere, please review this to make sure I covered the missing parts of the context variables. I'd also like your opinion on whether we should keep the `lookup` page separate or just collapse it into the new one. Thanks!

File: mule-user-guide/v/4.1/dataweave-runtime-functions.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+[source,xml, linenums]
+----
+<configuration-properties file="properties.yaml"/>
+
+<flow name="simple">
+  <logger message="#[p('some.property')]"/>
+</flow>
+----
+
+If an entry for `some.property` is present in `properties.yaml` then that will be
```
Comment (pablolagreca) [inline mule-user-guide/v/4.1/dataweave-runtime-functions.adoc]: I think this should reference the <configuration-properties> documentation section that describes the precedence for resolution. If not we are duplicating that docs here and they may get out of sync at some point.

Comment (pablolagreca) [reply mule-user-guide/v/4.1/dataweave-runtime-functions.adoc]: BTW: we need to check that this also works with secure::aProp

File: mule-user-guide/v/4.1/dataweave-variables-context.adoc [hunk 2]
```diff
@@ -35,8 +35,12 @@ DataWeave expressions accept variables that can retrieve values from a variety o
 |
 [%hardbreaks]
 //TODO
```
Comment (pablolagreca) [inline mule-user-guide/v/4.1/dataweave-variables-context.adoc]: remove

File: mule-user-guide/v/4.1/dataweave-variables-context.adoc [hunk 3]
```diff
@@ -35,8 +35,12 @@ DataWeave expressions accept variables that can retrieve values from a variety o
...
-//* Example: `_TODO_`
+* Fields:
+** `principal`: The main part of the authentication, such as the user or subject.
+** `credentials`: The credentials for the authentication.
+** `properties`: A Map of properties for the authentication.
+
+* Example: `#[authentication.principal]`
 
 |`correlationId`
 // Ask Dan Feist to provide description
```
Comment (pablolagreca) [inline mule-user-guide/v/4.1/dataweave-variables-context.adoc]: TODO or remove

File: mule-user-guide/v/4.1/dataweave-variables-context.adoc [hunk 4]
```diff
@@ -60,8 +64,15 @@ DataWeave expressions accept variables that can retrieve values from a variety o
 |
 [%hardbreaks]
 // See https://github.com/mulesoft/mule-api/blob/master/src/main/java/org/mule/runtime/api/message/Error.java
```
Comment (pablolagreca) [inline mule-user-guide/v/4.1/dataweave-variables-context.adoc]: TODO or remove

Comment (afelisatti) [reply mule-user-guide/v/4.1/dataweave-variables-context.adoc]: This is all from the docs team.

File: mule-user-guide/v/4.1/dataweave-runtime-functions.adoc (hunk 1 above)
Comment (afelisatti) [reply mule-user-guide/v/4.1/dataweave-runtime-functions.adoc]: will do

**Comment by dukesphere** [comment]
closing out, manually "merging" files into new dir: mule4-user-guide/v/4.1

**Pull request closed without merging**